### PR TITLE
Twitch HLS Update Fix (with quality settings and auth enabled)

### PR DESCRIPTION
**Pull request by MrSprigster**
Twitch HLS Update Fix (with quality settings and auth enabled)

### CODE REVIEW

**Comment by MrSprigster** [comment]
Hey guys,

In depth fix for the master repository version (rather than the version available xbmc)

Added fix for HLS streams
Added quality options to fix
Added auth (future proofing) - thanks to Kr0nZ for this suggestion.

Fork this Repo and Enjoy! :+1:

**Comment by derrod** [comment]
Works great.

**Comment by rickydeez** [comment]
Not too familiar with GitHub, how do I download a zip of these fixes to test?

**Comment by Giacom** [comment]
You can go [here](https://github.com/MrSprigster/Twitch.tv-on-XBMC), to the PR author's fork, and click on the "download zip" button on the middle-right side.

@ The Author: Maybe you should change the version number on addon.xml

**Comment by Giacom** [comment]
After some short testing it is working great so far. I'll see if any issues come up after I use it more.

If you want to test this then I recommend when you download the zip that you change the addon.xml to have a different name, it was causing me issues before but I don't think it will cause any when it is merged.

**Comment by MrSprigster** [comment]
Thanks for the testing and the addon.xml reminder @Giacom. 

@rickydeez, did you manage to find the zip download?

**Comment by rickydeez** [comment]
Yes - is seems to be working great, thanks!

File: addon.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
   <extension point='xbmc.addon.metadata'>
     <platform>all</platform>
     <language>en</language>
-    <source>https://github.com/StateOfTheArt89/Twitch.tv-on-XBMC</source>
+    <source>https://github.com/MrSprigster/Twitch.tv-on-XBMC</source>
```
Comment (Giacom) [inline addon.xml]: Um, is this a mistake? I think it should stay linking to this repo.

Comment (MrSprigster) [inline addon.xml]: I wasn't sure if this needed to point to my fork until it was merged. (thanks for querying this @giacom) - Reverted

**Comment by MrSprigster** [comment]
@rickydeez, Good to hear :)

**Comment by delta1** [comment]
Hi @StateOfTheArt89 when will this fix be uploaded to the addon repo?

**Comment by StateOfTheArt89** [comment]
I want to refractor the new code and test it again. After that i do a pull request for the addon repo (usually takes a few hours to be added to the repo). All in all it should be online this evening or tomorrow morning.

**Comment by MrSprigster** [comment]
Thanks @StateOfTheArt89, sorry for the messy code (including lack of error handling), my native language is C/C++ and I kinda learned the Python syntax as I was writing the fix.   I also wanted to get people (including myself) up and running ASAP as I use your inspired Addon on a daily basis and I'm sure there are many more who also do the same  :+1: 

@delta1, My fix for this isn't stable or clean enough for public release on the XBMC repository, it was just a temporary quick fix .

**Comment by MrSprigster** [comment]
You are welcome @uglypony :+1:. Enjoy!

**Comment by delta1** [comment]
@StateOfTheArt89 thank you so much for you work on this addon! It's really awesome :) 

@MrSprigster no worries, thank you for the contribution! unfortunately I can't update it manually with the fix since I'm on Openelec build of xmbc on Raspberry Pi.

**Comment by MrSprigster** [comment]
@delta1  The following link may be of use to you (should allow you to move the files over to your Pi from a shared folder on a windows PC)

http://www.xbmchub.com/forums/raspberry-pi-discussion/4956-%5Bhow-%5D-mount-windows-share-openelec-boot.html

Alternatively, use ssh command line and wget https://github.com/StateOfTheArt89/Twitch.tv-on-XBMC/archive/master.zip and then extract the zip to the correct folder on the Pi. (it may work by using the install from zip menu item in XBMC > Settings > Addons)

**Comment by StateOfTheArt89** [comment]
I did some code refactoring, but it is still far from perfect. There is still dead code (the whole rtmp stuff) and a few other issues. But the plugin seems to work nice for now, so I will go for a pull request at the xbmc repo. Thanks to @MrSprigster

**Comment by MrSprigster** [comment]
You are welcome :+1:. I'm super appreciative that this Addon exists! so a big thank you and lots of respect to you for making it exist in the first place.

**Comment by sgraef4587** [comment]
I am also very appreciative this Addon exists.  Thank you so much for every ounce of effort put into making and maintaining this Addon

**Pull request merged**